### PR TITLE
Submit autofill feedback report

### PR DESCRIPTION
**Pull request by CDRussell**
Task/Issue URL: https://app.asana.com/0/488551667048375/1207769724660967/f 

### Description
Handles the actual submission of the autofill breakage report.

### Steps to test this PR

ℹ️ Local hack to enable feature: In `AutofillSiteBreakageReportingFeature`, set default value to `true`
ℹ️ Logcat filter: `autofill_logins_report_failure` 

---

- [ ] Manually add a password for fill.dev
- [ ] Visit https://fill.dev
- [ ] Tap on overflow->Passwords
- [ ] Tap on `Report a problem with autofill` and choose to `Send Report`
- [ ] Verify logs contain `autofill_logins_report_failure`
- [ ] Verify params match up with your current state. e.g., if signed in to email protection, language code is picked up

### CODE REVIEW

**Comment by CDRussell** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4763?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4763** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4763?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4741** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4741?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4740** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4740?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @CDRussell and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt [hunk 1]
```diff
@@ -2525,9 +2528,16 @@ class BrowserTabFragment :
     }
 
     private fun launchAutofillManagementScreen() {
+        val privacyProtectionStatus = when (viewModel.privacyShieldViewState.value?.privacyShield) {
```
Comment (cmonfortep) [inline app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt:2531]: Can we make it that you receive this value as part of the command? That way viewmodel is the one accessing that value, not the view.

File: app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt [hunk 2]
```diff
@@ -2525,9 +2528,16 @@ class BrowserTabFragment :
...
+            PROTECTED -> PrivacyProtectionStatus.Enabled
```
Comment (cmonfortep) [inline app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt:2532]: I'm unsure this is the logic you want. I would say is wrong to rely on the privacy shield to decide if protections are enabled/disabled. http site is unprotected, but protections are enabled.

File: autofill/autofill-api/src/main/java/com/duckduckgo/autofill/api/AutofillScreens.kt [hunk 3]
```diff
@@ -56,3 +61,24 @@ enum class AutofillSettingsLaunchSource {
     InternalDevSettings,
     Unknown,
 }
+
+@Parcelize
+/**
+ * The status of the privacy protections for the current page
+ */
+sealed interface PrivacyProtectionStatus : Serializable, Parcelable {
```
Comment (cmonfortep) [inline autofill/autofill-api/src/main/java/com/duckduckgo/autofill/api/AutofillScreens.kt:69]: do we need a sealed class for this? Seems you are just trying to know if protections are enabled or not. Will it be enough with a boolean?

File: autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/impl/ui/credential/management/AutofillSettingsViewModel.kt [hunk 4]
```diff
@@ -152,6 +156,7 @@ class AutofillSettingsViewModel @Inject constructor(
     private var combineJob: Job? = null
 
     private var currentUrl: String? = null
+    private var privacyProtectionStatus: PrivacyProtectionStatus = Unknown
```
Comment (cmonfortep) [inline autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/impl/ui/credential/management/AutofillSettingsViewModel.kt:159]: ok, this is something similar to `currentUrl`. I think that should belong to the view state, and not having it here separated. Seems connected to the report, currentUrl, as a single thing, and we are keeping them independent which may be prone to errors.

**Review by cmonfortep — COMMENTED** [review]
Just some comments around privacy protection status, and how we are calculating that. and about unifying the viewmodel state into viewstate.

File: app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt (hunk 1 above)
Comment (CDRussell) [reply app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt]: As discussed offline, changed how this value is determined. Thanks!

File: app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt (hunk 2 above)
Comment (CDRussell) [reply app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt]: As discussed offline, changed how this value is determined. Thanks!

File: autofill/autofill-api/src/main/java/com/duckduckgo/autofill/api/AutofillScreens.kt (hunk 3 above)
Comment (CDRussell) [reply autofill/autofill-api/src/main/java/com/duckduckgo/autofill/api/AutofillScreens.kt]: Removed

File: autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/impl/ui/credential/management/AutofillSettingsViewModel.kt (hunk 4 above)
Comment (CDRussell) [reply autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/impl/ui/credential/management/AutofillSettingsViewModel.kt]: As discussed, wrapped up into a single object and stored in the view state

**Review by cmonfortep — APPROVED** [review]
LGTM

**Comment by CDRussell** [comment]
### Merge activity

* **Jul 24, 5:55 PM EDT**: @CDRussell started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/4763).

**Pull request merged**